### ISSUE TITLE
No role assigned

### ISSUE THREAD

**Issue by markward** [markward/enrol_autoenrol#7]
Samuel Than
Fri, 20 Feb 2015, 5:37 AM

There's somethign strange with this plugin, i've installed that on Moodle 2.7.3 and configure it to auto enrol on a specific course. the users's that click on the course auto-enrolled without problem.. HOWEVER, there was no student role or any role assigned...

Anything i've misconfigured ? 

Thanks for the help

**Comment by markward** [comment]
Hi Sam,

I haven't been able to reproduce this issue so far. I was wondering if you could answer a couple of questions for me?

1. When you check the settings for the enrolment method what do you see next to the "Role" setting (under "show more")?
2. Can you remember the process you took to set this instance up? Was it automatically added to a course or manually added?

Thanks very much

Mark

**Comment by samuelthan** [comment]
Responding to the question:
1. Student
2. Manually added. Meaning Auto-Enrol is available for all courses as invisible, i have to go to the particular course to enable/make visible.

**Comment by markward** [comment]
Hi Sam,

I've tested this scenario as far as I can and haven't been able to reproduce this so far.

Can I ask for some clarification around your mention that autoenrol is available for all courses as invisible. Do you mean it is added to all new courses by default but disabled (like the screenshot below)?

![Disabled instance](https://cloud.githubusercontent.com/assets/944391/6413876/6e1a32f6-be8b-11e4-8e50-0a5a593f1d49.png)

If so, can you tell me how that has been achieved? There is an option to add autoenrol to a course by default but not to have it disabled.

Many thanks

MArk

**Comment by samuelthan** [comment]
No. Let me clarify as i may have miscommunicated. Auto Enrol is NOT added to ALL courses by default.

Only selected courses that i wanted auto enrol to happen. I manually add the enrolment method via
Course administration > Users > Enrolment Method >     then add method auto enrol.
With that i would then get the screenshot as you've described.
